sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/Paginator.java

```diff
@@ -128,6 +134,18 @@
                 isChangeFeed,
                 top,
                 maxPageSize),
-                Queues.XS_BUFFER_SIZE);
+                preFetchCount);
+    }
+
+    public static int getPreFetchCount(CosmosQueryRequestOptions queryOptions, int top, int maxPageSize) {
+        int maxBufferedItemCount = queryOptions != null ? queryOptions.getMaxBufferedItemCount() : 0;
+        if (maxBufferedItemCount <= 0) {
+            return Queues.XS_BUFFER_SIZE;
+        }
+        int effectivePageSize = top > 0 ?
+            Math.min(Math.max(1, top), maxPageSize) :
```

Comment:
Math.min(top, maxPageSize)should be enough

Reply:
ACK -added the if later - missed to simplify. Done that now.